evalml/tests/component_tests/test_components.py

```diff
@@ -798,14 +798,18 @@ def test_all_transformers_check_fit(X_y_binary):
         component.transform(X)
 
 
-def test_all_estimators_check_fit(X_y_binary, test_estimator_needs_fitting_false, helper_functions):
-    X, y = X_y_binary
+def test_all_estimators_check_fit(X_y_binary, ts_data, test_estimator_needs_fitting_false, helper_functions):
     estimators_to_check = [estimator for estimator in _all_estimators() if estimator not in [StackedEnsembleClassifier, StackedEnsembleRegressor, TimeSeriesBaselineEstimator]] + [test_estimator_needs_fitting_false]
     for component_class in estimators_to_check:
         if not component_class.needs_fitting:
             continue
 
         component = helper_functions.safe_init_component_with_njobs_1(component_class)
+        if component.supported_problem_types == [ProblemTypes.TIME_SERIES_REGRESSION]:
```

Comment:
Prophet does not follow sklearn's estimator API (and this is our first estimator that only does time series problem types) and tests in `test_components.py` break so I've added some work arounds here for now. Happy to discuss any of these changes.

Reply:
Nice! I think it's fine to have this code outside a util function since it's only used twice and only in our tests.